sdk/storage/azure-storage-file-datalake/samples/datalake_samples_access_control_recursive.py

```diff
@@ -0,0 +1,112 @@
+    1) STORAGE_ACCOUNT_NAME - the storage account name
+    2) STORAGE_ACCOUNT_KEY - the storage account key
+"""
+
+import os
+import random
+import uuid
+
+from azure.storage.filedatalake import (
+    DataLakeServiceClient,
+)
+
+
+def recursive_access_control_sample(filesystem_client):
+    # create a parent directory
+    dir_name = "testdir"
+    print("Creating a directory named '{}'.".format(dir_name))
+    directory_client = filesystem_client.create_directory(dir_name)
+
+    # populate the directory with some child files
+    create_child_files(directory_client, 35)
+
+    # get and display the permissions of the parent directory
+    acl_props = directory_client.get_access_control()
+    print("Permissions of directory '{}' are {}.".format(dir_name, acl_props['permissions']))
+
+    # set the permissions of the entire directory tree recursively
+    # update/remove acl operations are performed the same way
+    acl = 'user::rwx,group::r-x,other::rwx'
+    failed_entries = []
+
+    # the progress callback is invoked each time a batch is completed
+    def progress_callback(resp):
+        print("In this batch: {} directories and {} files were processed successfully, {} failures were counted"
+              .format(resp.directories_successful, resp.files_successful, resp.failure_count))
+
+        # keep track of failed entries if there are any
+        failed_entries.append(resp.failed_entries)
+
+    # illustrate the operation by using a small batch_size
+    summary = directory_client.set_access_control_recursive(acl=acl, progress_callback=progress_callback,
+                                                            batch_size=5)
+    print("Summary: {} directories and {} files were updated successfully, {} failures were counted."
+          .format(summary.directories_successful, summary.files_successful, summary.failure_count))
+
+    # if an error was encountered, a continuation token would be returned if the operation can be resumed
+    if summary.continuation is not None:
+        print("The operation can be resumed by passing the continuation token {} again into the access control method."
+              .format(summary.continuation_token))
+
+    # get and display the permissions of the parent directory again
+    acl_props = directory_client.get_access_control()
+    print("New permissions of directory '{}' and its children are {}.".format(dir_name, acl_props['permissions']))
+
+
+def create_child_files(directory_client, num_child_files):
+    import concurrent.futures
+    import itertools
+    # Use a thread pool because it is too slow otherwise
+    with concurrent.futures.ThreadPoolExecutor(max_workers=10) as executor:
+        def create_file():
+            # generate a random name
+            file_name = str(uuid.uuid4()).replace('-', '')
+            directory_client.get_file_client(file_name).create_file()
+
+        futures = {executor.submit(create_file) for _ in itertools.repeat(None, num_child_files)}
```

Comment:
>{ [](start = 18, length = 1)

curious why we are using { instead of [ here?

Reply:
I got it from the directory sample